bin/all_sky_search/pycbc_upload_single_event_to_gracedb

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python
+
+# Copyright (C) 2015-2023 Ian Harry, Gareth Cabourn Davies
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+"""
+Take a coinc xml file containing a single event and upload to gracedb.
+"""
+
+import os
+import sys
+import argparse
+import logging
+from ligo.gracedb.rest import GraceDb
+import numpy as np
+
+import pycbc
+import lal
+from pycbc.io.live import gracedb_tag_with_version
+
+parser = argparse.ArgumentParser(description=__doc__)
+parser.add_argument('--xml-file-for-upload', required=True, type=str,
+                    help='LIGOLW XML file containing the event.')
+parser.add_argument('--log-message', type=str, metavar='MESSAGE',
+                    help='Add a log entry to each upload with the given message')
+parser.add_argument('--testing', action="store_true", default=False,
+                    help="Upload event to the TEST group of gracedb.")
+parser.add_argument('--production-server', action="store_true", default=False,
+                    help="Upload event to production graceDB. If not given "
+                         "events will be uploaded to playground server.")
+parser.add_argument('--force-overwrite', action='store_true', default=False,
+                    help="GraceDb instance will be checked for if an event "
+                         "with the same event time and FAR already exist. "
+                         "If so, event will not be uploaded")
```

Comment:
The name of the option seems inconsistent with the help string.

Reply:
…wait, this option does not seem to be used in the code.

Reply:
Same for --query-string. Are those inherited from old code?

Reply:
Inherited - I will remove